x-pack/filebeat/input/httpjson/value_tpl.go

```diff
@@ -295,6 +303,100 @@ func div(a, b int64) int64 {
 	return a / b
 }
 
+// minInt returns the minimum of a and an arbitrary number of integers.
+func minInt(a int, nums ...int) int {
+	min := math.MaxInt
+	nums = append(nums, a)
+	if len(nums) == 1 {
+		return nums[0]
+	}
+	for _, num := range nums {
+		min = findMin(min, num)
+	}
+	return min
+}
```

Comment:
These all allocate. I think this is a case where a generic definition would be best.
```
type number interface { ~int | ~uint | ~float64}

func min[T number](head T, tail ..T) T {
    for _, t := range tail {
        if t != t {
            return t
        }
        if t < head {
            head = t
        }
    }
    return head
}

func max[T number](head T, tail ..T) T {
    for _, t := range tail {
        if t != t {
            return t
        }
        if t > head {
            head = t
        }
    }
    return head
}
```
Then for the map
```
			"maxInt":              max[int],
			"minInt":              min[int],
			"maxUint":             max[uint],
			"minUint":             min[uint],
			"maxFloat":            max[float64],
			"minFloat":            min[float64],
```

Reply:
If we go with this approach then we are once again lumping values of different ranges and signs together, are we not ?

Reply:
Yes this seems to be a good approach